### PR TITLE
other(google-sheets): add a constraint on row index

### PR DESCRIPTION
**Pull request by mathias-vandaele**
## Description

Fix a small issue found by @slolatte 

Now the row index field is restricted to either a FEEL expression, a number or empty String

### CODE REVIEW

File: connectors/google/google-sheets/element-templates/google-sheets-outbound-connector.json [hunk 1]
```diff
@@ -443,6 +443,12 @@
     "label" : "Row index",
     "description" : "Enter row index. Details in the <a href=\"https://docs.camunda.io/docs/components/connectors/out-of-the-box-connectors/google-sheets/#what-is-a-row-index\" target=\"_blank\">documentation</a>",
     "optional" : true,
+    "constraints" : {
+      "notEmpty" : false,
+      "pattern" : {
+        "value" : "^(=.*|[0-9].*|)$"
```
Comment (johnBgood) [inline connectors/google/google-sheets/element-templates/google-sheets-outbound-connector.json:449]: I think it's not correct, `9abc` would be a valid input using this. Maybe should you change the digit part to [0-9]+ or \d+

File: connectors/google/google-sheets/element-templates/google-sheets-outbound-connector.json [hunk 2]
```diff
@@ -443,6 +443,12 @@
     "label" : "Row index",
     "description" : "Enter row index. Details in the <a href=\"https://docs.camunda.io/docs/components/connectors/out-of-the-box-connectors/google-sheets/#what-is-a-row-index\" target=\"_blank\">documentation</a>",
     "optional" : true,
+    "constraints" : {
+      "notEmpty" : false,
```
Comment (johnBgood) [inline connectors/google/google-sheets/element-templates/google-sheets-outbound-connector.json:447]: is this added by the template generator?

Comment (mathias-vandaele) [reply connectors/google/google-sheets/element-templates/google-sheets-outbound-connector.json:447]: yep

**Review by johnBgood — DISMISSED** [review]

File: connectors/google/google-sheets/src/main/java/io/camunda/connector/gsheets/model/request/input/CreateRow.java [hunk 3]
```diff
@@ -43,9 +42,11 @@ public record CreateRow(
                 "Enter row index. Details in the <a href=\"https://docs.camunda.io/docs/components/connectors/out-of-the-box-connectors/google-sheets/#what-is-a-row-index\" target=\"_blank\">documentation</a>",
             group = "operationDetails",
             feel = FeelMode.optional,
+            constraints =
+                @PropertyConstraints(
+                    pattern = @TemplateProperty.Pattern(value = "^(=.*|[0-9]+|)$", message = "")),
```
Comment (johnBgood) [inline connectors/google/google-sheets/src/main/java/io/camunda/connector/gsheets/model/request/input/CreateRow.java:47]: We could add a message here

**Review by sbuettner — APPROVED** [review]

**Pull request merged**